### PR TITLE
EVA-2175 - Validation process to reconcile RS IDs in the release against those in Mongo

### PR DESCRIPTION
**Pull request by sundarvenkata-EBI**
**eva-accession-release-automation/run_release_in_embassy/validate_rs_release_files.py**
* Retrieves RS IDs from Mongo and release files, compares them and makes attribution to the RS IDs that have not been released. The [Mongo queries for missing ID attributions remain the same](https://www.ebi.ac.uk/panda/jira/secure/attachment/36493/Validate_script_comparison.html) as the ones used in the previous release but the rest of the script is heavily modified. See usage details [here](https://www.ebi.ac.uk/panda/jira/browse/EVA-2157?focusedCommentId=338260&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-338260).

**eva-accession-release-automation/run_release_in_embassy/analyze_vcf_validation_files.py**
* Analyze VCF validator and assembly report check logs to ensure that they don't have any errors that are not already expected during the checks.

### CODE REVIEW

**Review by andresfsilva — APPROVED** [review]

**Comment by sundarvenkata-EBI** [comment]
Added another [commit](https://github.com/EBIvariation/eva-accession/pull/243/commits/5feb0496868151b6917fdb60735ae4e4929d3165) based on the warning observed when processing mouse

File: eva-accession-release-automation/run_release_in_embassy/requirements.txt [hunk 1]
```diff
@@ -1,2 +1,3 @@
 ebi_eva_common_pyutils==0.3.4
```
Comment (tcezard) [inline eva-accession-release-automation/run_release_in_embassy/requirements.txt:1]:
```suggestion
ebi_eva_common_pyutils==0.3.5
```

File: eva-accession-release-automation/run_release_in_embassy/analyze_vcf_validation_results.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import click
+import glob
+import logging
+import sys
+
+from ebi_eva_common_pyutils.command_utils import run_command_with_output
```
Comment (tcezard) [inline eva-accession-release-automation/run_release_in_embassy/analyze_vcf_validation_results.py:22]:
```suggestion
from ebi_eva_common_pyutils.command_utils import run_command_with_output
from ebi_eva_common_pyutils.logger import logging_config as log_cfg
```

File: eva-accession-release-automation/run_release_in_embassy/analyze_vcf_validation_results.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+from run_release_in_embassy.release_metadata import vcf_validation_output_file_pattern, asm_report_output_file_pattern
+
+logger = logging.getLogger(__name__)
```
Comment (tcezard) [inline eva-accession-release-automation/run_release_in_embassy/analyze_vcf_validation_results.py:25]:
```suggestion
logger = log_cfg.get_logger(__name__)
```

File: eva-accession-release-automation/run_release_in_embassy/analyze_vcf_validation_results.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+                                                     "do not share the first nucleotide",
+                                                     "the input file is not valid",
+                                                     "the input file is valid",
+                                                     "not listed in a valid meta-data ALT entry"]
+    vcf_validation_error_grep_command_chain = " | ".join(['grep -v "{0}"'.format(error_class) for error_class in
+                                                          vcf_validation_report_error_classes_to_ignore])
+    for vcf_validation_report_file in vcf_validation_report_files:
+        logger.info("Analyzing file {0} ....".format(vcf_validation_report_file))
+        command_to_run = "cat {0} | {1} | wc -l".format(vcf_validation_report_file,
+                                                        vcf_validation_error_grep_command_chain)
```
Comment (tcezard) [inline eva-accession-release-automation/run_release_in_embassy/analyze_vcf_validation_results.py:41]: I'm not sure it's better or more efficient, but I prefer `grep -P`

```suggestion
    vcf_validation_error_grep = "|".join([vcf_validation_report_error_classes_to_ignore])
    for vcf_validation_report_file in vcf_validation_report_files:
        logger.info("Analyzing file {0} ....".format(vcf_validation_report_file))
        command_to_run = "grep -cvP '{0}' {1}".format(vcf_validation_error_grep, vcf_validation_report_file)
```

File: eva-accession-release-automation/run_release_in_embassy/analyze_vcf_validation_results.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+def analyze_asm_report_files(asm_report_files):
+    exit_code = 0
+    assembly_report_error_classes_to_ignore = ["not present in FASTA file", "does not match the reference sequence"]
+
+    asm_report_error_grep_command_chain = " | ".join(['grep -v "{0}"'.format(error_class) for error_class in
+                                                      assembly_report_error_classes_to_ignore])
+    for asm_report_file in asm_report_files:
+        logger.info("Analyzing file {0} ....".format(asm_report_file))
+        command_to_run = "cat {0} | {1} | wc -l".format(asm_report_file, asm_report_error_grep_command_chain)
```
Comment (tcezard) [inline eva-accession-release-automation/run_release_in_embassy/analyze_vcf_validation_results.py:60]: Same as above
```suggestion
    asm_report_error_grep = "|".join(assembly_report_error_classes_to_ignore)
    for asm_report_file in asm_report_files:
        logger.info("Analyzing file {0} ....".format(asm_report_file))
        command_to_run = "grep -cvP '{0}' {1}".format(asm_report_error_grep, asm_report_file)
```

**Review by tcezard — APPROVED** [review]
Left a few comments none of which are blocking

File: eva-accession-release-automation/run_release_in_embassy/analyze_vcf_validation_results.py (hunk 2 above)
Comment (sundarvenkata-EBI) [reply eva-accession-release-automation/run_release_in_embassy/analyze_vcf_validation_results.py:22]: Addressed

File: eva-accession-release-automation/run_release_in_embassy/requirements.txt (hunk 1 above)
Comment (sundarvenkata-EBI) [reply eva-accession-release-automation/run_release_in_embassy/requirements.txt]: Addressed

File: eva-accession-release-automation/run_release_in_embassy/analyze_vcf_validation_results.py (hunk 3 above)
Comment (sundarvenkata-EBI) [reply eva-accession-release-automation/run_release_in_embassy/analyze_vcf_validation_results.py]: Addressed

File: eva-accession-release-automation/run_release_in_embassy/analyze_vcf_validation_results.py (hunk 4 above)
Comment (sundarvenkata-EBI) [reply eva-accession-release-automation/run_release_in_embassy/analyze_vcf_validation_results.py:42]: I am not very well versed with PCRE regexps and I am wary of semicolons in the text being interpreted differently. So I would like to leave the current command chain as it is.

File: eva-accession-release-automation/run_release_in_embassy/analyze_vcf_validation_results.py (hunk 5 above)
Comment (sundarvenkata-EBI) [reply eva-accession-release-automation/run_release_in_embassy/analyze_vcf_validation_results.py:61]: See [comment](https://github.com/EBIvariation/eva-accession/pull/243#discussion_r500133453)

**Pull request merged**